### PR TITLE
feat(route/mirrormedia): Add route for category.

### PR DESCRIPTION
**Pull request by dzx-dzx**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/mirrormedia/category/political
/mirrormedia/section/entertainment
/mirrormedia
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/mirrormedia/filter.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            variables: {
+                take: ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 24,
+                skip: 0,
+                orderBy: { publishedDate: 'desc' },
+                filter: {
+                    state: { equals: 'published' },
+                    categories: { some: { slug: { equals: category } } },
+                },
+            },
+            query: 'fragment section on Section {\n  id\n  name\n  slug\n  state\n  __typename\n}\n\nfragment category on Category {\n  id\n  name\n  slug\n  state\n  __typename\n}\n\nfragment listingPost on Post {\n  id\n  slug\n  title\n  brief\n  publishedDate\n  state\n  sections(where: {state: {equals: "active"}}) {\n    ...section\n    __typename\n  }\n  categories(where: {state: {equals: "active"}}) {\n    ...category\n    __typename\n  }\n  isFeatured\n  __typename\n}\n\nquery ($take: Int, $skip: Int, $orderBy: [PostOrderByInput!]!, $filter: PostWhereInput!) {\n  postsCount(where: $filter)\n  posts(take: $take, skip: $skip, orderBy: $orderBy, where: $filter) {\n    ...listingPost\n    __typename\n  }\n}',
```
Comment (TonyRL) [inline lib/routes/mirrormedia/filter.ts:38]: https://docs.rsshub.app/joinus/advanced/script-standard#string
> Use [template literals](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Template_literals) for GraphQL queries as they make the code more concise and easy to read.

Comment (dzx-dzx) [reply lib/routes/mirrormedia/filter.ts:38]: But the query string is constant?

Comment (TonyRL) [reply lib/routes/mirrormedia/filter.ts:38]: Most likely yes

File: lib/routes/mirrormedia/category.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            variables: {
+                take: ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 24,
+                skip: 0,
+                orderBy: { publishedDate: 'desc' },
+                filter: {
+                    state: { equals: 'published' },
+                    ...categoryFilter, ...sectionFilter,
+                },
+            },
+            query: `fragment section on Section {\n  id\n  name\n  slug\n  state\n  __typename\n}\n\nfragment category on Category {\n  id\n  name\n  slug\n  state\n  __typename\n}\n\nfragment listingPost on Post {\n  id\n  slug\n  title\n  brief\n  publishedDate\n  state\n  sections(where: {state: {equals: "active"}}) {\n    ...section\n    __typename\n  }\n  categories(where: {state: {equals: "active"}}) {\n    ...category\n    __typename\n  }\n  isFeatured\n  __typename\n}\n\nquery ($take: Int, $skip: Int, $orderBy: [PostOrderByInput!]!, $filter: PostWhereInput!) {\n  postsCount(where: $filter)\n  posts(take: $take, skip: $skip, orderBy: $orderBy, where: $filter) {\n    ...listingPost\n    __typename\n  }\n}`,
```
Comment (TonyRL) [inline lib/routes/mirrormedia/category.ts:40]: [Template literals](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Template_literals) can include [line breaks](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Template_literals#string_text).

**Pull request merged**